src/compiler/checker.ts

```diff
@@ -18595,7 +18595,19 @@ namespace ts {
             // Grammar checking
             checkGrammarStatementInAmbientContext(node);
 
-            checkExpression(node.expression);
+            const type = checkExpression(node.expression);

```

Comment:
I mean you check only `ExpressionStatement`
```ts
function checkExpressionStatement(node: ExpressionStatement)
```
So this example has a missed `await`, but the error doesn't happen, because it is not `ExpressionStatement`, it's a `ReturnStatement`
```ts
async function bar(){
   return isAsync();
}
```